src/librustdoc/html/static/main.js

```diff
@@ -903,12 +901,10 @@ function defocusSearchBar() {
                 // Names didn't match so let's check if one of the generic types could.
                 if (literalSearch === true) {
                      if (obj.length > GENERICS_DATA && obj[GENERICS_DATA].length > 0) {
-                        var length = obj[GENERICS_DATA].length;
-                        for (x = 0; x < length; ++x) {
-                            if (obj[GENERICS_DATA][x] === val.name) {
-                                return true;
-                            }
-                        }
+                        return onEach(obj[GENERICS_DATA],
+                            function(name) {
+                                return name === val.name;
+                            });
```

Comment:
```suggestion
                        return obj[GENERICS_DATA].any(
                            function(name) {
                                return name === val.name;
                            });
```

Reply:
You mean `some` I guess? ;)

Reply:
And some is indeed faster! I tested on jsbench again the two codes:

```js
function onEach(arr, func) {
	for (var i = 0, len = arr.length; i < len; ++i) {
		if (func(arr[i]) === true) {
			return true;
		}
	}
	return false;
}

var arr = ["a", "bcdef", "ghi", "ksdf", "jsqdfkqsdfq", "qdjkfqdq", "dfkqjdnfq", "qdjkfqdjkfq", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd",
"qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd",
"qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd",
"qfqdjfnqd", "dada"];
var found = onEach(arr, function(el) { return el === "dada"});
```

and:

```js
var arr = ["a", "bcdef", "ghi", "ksdf", "jsqdfkqsdfq", "qdjkfqdq", "dfkqjdnfq", "qdjkfqdjkfq", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd",
"qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd",
"qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd", "qfqdjfnqd",
"qfqdjfnqd", "dada"];
var found = arr.some(function(el) { return el === "dada"});
```

The first one is ~8.47 slower. So let's go for `some` and thanks a lot for the suggestion!